Content.Server/GameObjects/Components/BaseComputerUserInterfaceComponent.cs

```diff
@@ -0,0 +1,144 @@
+using Content.Server.GameObjects.Components.Construction;
+using Content.Server.GameObjects.Components.Power.ApcNetComponents;
+using Content.Server.Utility;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.GameObjects.EntitySystems;
+using Content.Shared.GameObjects.EntitySystems.ActionBlocker;
+using Content.Shared.Interfaces;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.ViewVariables;
+using Robust.Shared.Localization;
+
+namespace Content.Server.GameObjects.Components
+{
+    /// <summary>
+    /// This component is used as a base class for classes like SolarControlConsoleComponent.
+    /// These components operate the server-side logic for the "primary UI" of a computer.
+    /// That means showing the UI when a user activates it, for example.
+    /// </summary>
+    public abstract class BaseComputerUserInterfaceComponent : Component
```

Comment:
We're generally moving away from component inheritance. The problem is now every single component that wants to only require power for interactions now inherits this which is very dirty.

Better off making it its own component to require power for the boundinterface or something.

Reply:
That isn't tenable due to the situations where overrides are required.
The problem is, *without* overrides we're left with the present situation where everything needs too much copy/pasting.
Inheritance really is the best way I can think of here.